### ISSUE TITLE
Removal of AngularJS support in Grafana

### ISSUE THREAD

**Issue by marmila** [ricsanfre/pi-cluster#383]
Hello,

I recently came across an important update from Grafana, detailed in a blog post on their official site ([Removal of AngularJS Support in Grafana: What You Need to Know](https://grafana.com/blog/2024/03/11/removal-of-angularjs-support-in-grafana-what-you-need-to-know/)), announcing the removal of AngularJS support. Given that this project relies on Grafana for visualizations/dashboards, this development might significantly impact its functionality or future compatibility.

To summarize, the post outlines that starting from Grafana version 11 (the specific version where AngularJS support is removed), dashboards or plugins developed with AngularJS will no longer function as expected without migration to a supported framework (e.g., React).  I think it's an important discussion we need to start. I'm also more than willing to assist in this process, whether it be through research, testing, or development.

Thx in advance.

**Comment by ricsanfre** [comment]
Hi Marco,

Thanks for the heads-up.

Most of the dashboards come from kube-prometheus-stack helm chart, which also takes the dashboards from kubernetes-mixin project. There is an open issue, https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/898, to fix this AngularJS deprecation.
Anyway we are maintaining locally a set of dashboards, most of them from grafana.com. The list of local dashboards is in  https://github.com/ricsanfre/pi-cluster/tree/master/argocd/system/monitoring/dashboards.

I also have documented where to download the source json documents and what modifications I did on them. All documented here  https://picluster.ricsanfre.com/docs/prometheus/

Happy to accept any PR from you to help with this issue.

Regards
Ricardo.

**Comment by ricsanfre** [comment]
https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/898, has been already closed and changes merged into kube-prometheus-stack. See https://github.com/prometheus-community/helm-charts/issues/4380.

kubernetes dashboards locally maintained are now located in https://github.com/ricsanfre/pi-cluster/tree/master/kubernetes/platform/grafana/config/base/dashboards.

Possible way to automatize the generation of k3s local dashboards and alerts rules from original kubernetes-mixin dashborads here: https://hodovi.cc/blog/configuring-kube-prometheus-stack-dashboards-and-alerts-for-k3s-compatibility/

**Comment by ricsanfre** [comment]
Automated the generation of k3s mixins with commit 35457be1684e2cd486c56308e4cff9c5881773fd
